Plugins/ScriptLoader/Source/ScriptLoader.cpp

```diff
@@ -1,53 +1,53 @@
 #include "ScriptLoader.h"
 
-#include <Babylon/NetworkUtils.h>
+#include <UrlLib/UrlLib.h>
 #include <arcana/threading/task.h>
 
 namespace Babylon
 {
-    struct ScriptLoader::Impl
+    class ScriptLoader::Impl
     {
-        Impl(DispatchFunctionT dispatchFunction, std::string rootUrl)
-            : DispatchFunction{std::move(dispatchFunction)}
-            , RootUrl{std::move(rootUrl)}
-            , Task{arcana::task_from_result<std::exception_ptr>()}
+    public:
+        Impl(DispatchFunctionT dispatchFunction)
+            : m_dispatchFunction{dispatchFunction}
+            , m_task{arcana::task_from_result<std::exception_ptr>()}
         {
         }
 
         void LoadScript(std::string url)
         {
-            auto absoluteUrl = GetAbsoluteUrl(url, RootUrl);
-            auto loadUrlTask = LoadTextAsync(std::move(absoluteUrl));
-            arcana::task_completion_source<void, std::exception_ptr> completionSource{};
-            auto loadScriptTask = arcana::when_all(loadUrlTask, Task).then(arcana::inline_scheduler, arcana::cancellation::none(), [dispatchFunction = DispatchFunction, url = std::move(url), completionSource](const std::tuple<std::string, arcana::void_placeholder>& args) mutable {
-                std::string source = std::get<0>(args);
-                dispatchFunction([source = std::move(source), url = std::move(url), completionSource = std::move(completionSource)](Napi::Env env) mutable {
-                    Napi::Eval(env, source.data(), url.data());
-                    completionSource.complete();
+            UrlLib::UrlRequest request;
+            request.Open(UrlLib::UrlMethod::Get, url);
+            request.ResponseType(UrlLib::UrlResponseType::String);
+            m_task = arcana::when_all(m_task, request.SendAsync()).then(arcana::inline_scheduler, arcana::cancellation::none(), [dispatchFunction{m_dispatchFunction}, request{std::move(request)}, url{std::move(url)}](auto) {
```

Comment:
Not your change, but why is this `when_all` instead of just `m_task = m_task.then(...)`?

Reply:
Actually, if you chase this back, it is my change :-)

This is a when_all because loading the url should happen in parallel to loading the existing script (if an existing script is still loading).